### ISSUE TITLE
avoid the name clash with Prelude’s null

### ISSUE THREAD

**Issue by int-index** [zyla/haskell-nullpointer#1]
C++ uses `nullptr` for null pointer literals: https://en.cppreference.com/w/cpp/language/nullptr

This package could do the same to avoid a name clash. Or not. Just an idea.

**Comment by zyla** [comment]
Good idea. However, I think users could be unhappy if we dropped Java compatibility. What about overloading `null` so that it behaves like `Prelude.null` at function types, and like `Data.NullPointer.null` at other types?

**Comment by aloiscochard** [comment]
Yes, Java compatibility is a must here.

I propose to ask for a rename of `Prelude.null`, next year on 1st of April.
